posts/2024-07-26-actuated-for-gitlab.md

```diff
@@ -0,0 +1,217 @@
+---
+title: Secure CI for GitLab with Actuated.
+description: Learn how Actuated can improve your GitLab CI.
+tags:
+  - gitlab
+author_img: welteki
+date: "2024-07-26"
+---
+
+Last year we introduced a [tech preview for Actuated for GitLab CI](/blog/secure-microvm-ci-gitlab). Actuated was created to offer low maintenance, secure self-hosted CI runners that make optimal use of your available compute resources.
+
+Since the initial preview a lot of features have been added. In this article we will give you an overview of some of the available features and how they can benefit your GitLab CI.
+
+- [Introduction](#introduction)
+- [Run jobs in microVMs with Actuated](#run-jobs-with-actuated)
+- [Mixed docker and shell executors](#mixed-docker-and-shell-executors)
+- [Private peering](#private-peering)
+
+## Introduction
+
+It can be challenging to run GitLab CI/CD jobs that build and publish Docker images or jobs that require extensive system access in a safe way. Docker-in-Docker (DIND) requires the docker executor to run containers in privileged mode. Using the shell executor would give a job full access to the runner host and network. They can also leave behind side effects between builds as the runner is reused.
+
+Using both approaches causes a significant security concern and the [GitLab runner security docs](https://docs.gitlab.com/runner/security/) warn against it.
+
+![Docker in Docker security notice printed out by the GitLab helm chart](/images/2024-07-gitlab/chart-warning.png)
+> Security notice displayed by the GitLab Helm chart to explain why docker in docker is disabled by default for security purposes.
+
+With Actuated, jobs run in ephemeral microVMs using Linux KVM for secure isolation. After the job is completed, the VM will be destroyed and removed from the GitLab instance. This allows us to safely run DIND and the shell executor in a fresh isolated environment for each job.
+
+There are no horrible Kernel tricks or workarounds required to be able to use user namespaces, no need to change your tooling from what developers love - Docker, to Kaninko or Buildah or similar. You have `sudo` access and full VM with systemd available, things like Kubernetes will also work out of the box if you need them for end to end testing.
+
+![GitLab UI showing Actuated project runners](/images/2024-07-gitlab/project-runners.png)
+> Runners get automatically added to a project and are removed again when they finish running a job.
+
+## Run jobs in microVMs with Actuated
```

Comment:
`## Requirements for Actuated servers`

Explain what is required, since this question comes up a lot. Stress how it's just a bare-OS installation, supports overprovisioning, uses industry standard Firecracker and Cloud Hypervisor (if GPUs are required) and how easy it is to add machines as required. etc